appium/lib/screen-object.rb

```diff
@@ -66,82 +69,136 @@ def wait_step(timeout = 5, message = nil, &block)
+          when :down then  [x, y, x, y * 0.5, duration]
+          when :left then  [x * 0.6, y, x * 0.3, y, duration]
+          when :right then [x * 0.3, y, x * 0.6, y, duration]
+          else
+            raise('Only up, down, left and right scrolling are supported')
+          end
+    gesture(loc)
+  end
+
+  def scroll_down
+    scroll(:down)
+  end
+
+  def scroll_up
+    scroll(:up)
+  end
+
+  def swipe_left
+    scroll(:left)
+  end
+
+  def swipe_right
+    scroll(:right)
+  end
+
+  # Scroll Down in a direction until a string that matches is found,
+  #
+  # @param text      [String] The text you are looking for on the screen
+  def scroll_down_to_text(text)
+    scroll_text_to_view(text)
+  end
+
+  # Scroll Up in a direction until a string that matches is found,
+  #
+  # @param text      [String] The text you are looking for on the screen
+  def scroll_up_to_text(text)
+    scroll_text_to_view(text, :up)
+  end
+
+  # Scrolls in a direction until a string that matches is found,
+  #
+  # @param text      [String] The text you are looking for on the screen
+  # @param direction [symbol] The direction to search for an string
+  # @param timeout   [Integer] The amount of times in seconds we want to scroll to find the element
+  # @return          [Boolean]
+  def scroll_text_to_view(text, direction = :down, timeout = 40)
+    wait_until(timeout,'Unable to find element',&->{text_visible?(text, direction)})
+  end
+
+  # Scrolls in a direction if a text that matches is not found. return false,  otherwise return true
+  # Some locators on ios and android return true/false but a few would generate and error.
+  # this is the reason why there is a else condition and a rescue.
+  # @param text       [String] The text you are looking for on the screen
+  # @param direction [symbol] The direction to search for an string
+  # @return          [Boolean]
+  def text_visible?(text, direction)
+    if driver.find(text).displayed?
+      scroll(direction)
+      true
+    else
+      scroll(direction)
+      false
     end
+  rescue Selenium::WebDriver::Error::NoSuchElementError
+    scroll(direction)
+    false
   end
 
-  def scroll_down_click(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          driver.find_element(locator,locator_value).click
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollStart,:end_x =>0,:end_y =>scrollEnd,:touchCount => 2,:duration => 0)
-        false
-      end
+  # Find the first element containing value
+  # @param value [String] the value to search for
+  # @return [Element]
+  def get_element_by_text(value)
```

Comment:
Since `get_element_by_text` is related to element, let's move it to `element.rb`